### PR TITLE
feat(socialMediaLinks): add platform property for aria-label

### PR DESCRIPTION
**Pull request by mhayesm**
Platform property added to social links in the showcase and footer. This allows us to have 'Platform Title' in the aria label so the screen reader would read out 'Facebook @stockportMBC', whilst maintaining the Title text that's displayed on the page.

### CODE REVIEW

**Review by jordan-burns — APPROVED** [review]

File: src/StockportWebapp/Models/Showcase.cs [hunk 1]
```diff
@@ -44,6 +44,8 @@ public class Showcase
 
         public News NewsArticle { get; set; }
         public IEnumerable<Event> Events { get; set; }
+        public string Platform { get; set; }
```
Comment (lydiagray) [inline src/StockportWebapp/Models/Showcase.cs:47]: Should this not be part of showcase.SocialMediaLinks not it's own property

File: src/StockportWebapp/ProcessedModels/ProcessedShowcase.cs [hunk 2]
```diff
@@ -44,6 +44,8 @@ public class ProcessedShowcase : IProcessedContentType
         public readonly Video Video;
         public readonly string TypeformUrl;
         public readonly SpotlightBanner SpotlightBanner;
+        public readonly string Platform;
```
Comment (lydiagray) [inline src/StockportWebapp/ProcessedModels/ProcessedShowcase.cs:47]: Should this not be part of showcase.SocialMediaLinks not it's own property

File: src/StockportWebapp/ProcessedModels/ProcessedShowcase.cs [hunk 3]
```diff
@@ -86,7 +88,8 @@ public ProcessedShowcase()
             string eventsReadMoreText,
             Video video,
             string typeformUrl,
-            SpotlightBanner spotlightBanner)
+            SpotlightBanner spotlightBanner,
+            string platform)
```
Comment (lydiagray) [inline src/StockportWebapp/ProcessedModels/ProcessedShowcase.cs:92]: Should this not be part of showcase.SocialMediaLinks not it's own property

File: src/StockportWebapp/ProcessedModels/ProcessedShowcase.cs [hunk 4]
```diff
@@ -127,6 +130,7 @@ public ProcessedShowcase()
             Video = video;
             TypeformUrl = typeformUrl;
             SpotlightBanner = spotlightBanner;
+            Platform = platform;
```
Comment (lydiagray) [inline src/StockportWebapp/ProcessedModels/ProcessedShowcase.cs:133]: Should this not be part of showcase.SocialMediaLinks not it's own property

File: src/StockportWebapp/Views/Shared/SocialMediaLinks.cshtml [hunk 5]
```diff
@@ -18,7 +18,7 @@
         <a href="@socialMediaLink.Url">
             <span class="visuallyhidden">link to @nameOfSite @socialMediaLink.Title </span>
```
Comment (lydiagray) [inline src/StockportWebapp/Views/Shared/SocialMediaLinks.cshtml:19]: The change needed to be here, not on the span below

File: src/StockportWebapp/Views/stockportgov/Shared/Semantic/Showcase/SocialMediaLinks.cshtml [hunk 6]
```diff
@@ -3,12 +3,12 @@
...
-    {
-        <a as-link="true" class="showcase-button" href="@link.Url">
-            <span class="@link.Icon fa 2x" aria-hidden="true"></span>
-            <span>@link.Title</span>
-        </a>
-    }
+        @foreach (var link in Model.SocialMediaLinks)
+        {
+            <a as-link="true" class="showcase-button" href="@link.Url">
+                <span class="@link.Icon fa 2x" aria-label="@link.Platform @link.Title"></span>
```
Comment (lydiagray) [inline src/StockportWebapp/Views/stockportgov/Shared/Semantic/Showcase/SocialMediaLinks.cshtml:9]: The properties need to be the other way round. Title first then account name not platform

File: src/StockportWebapp/Models/SocialMediaLink.cs [hunk 7]
```diff
@@ -6,21 +6,23 @@ public class SocialMediaLink
         public string Slug { get; }
         public string Url { get; }
         public string Icon { get; }
+        public string Platform { get; }
 
-        public SocialMediaLink(string title, string slug, string url, string icon)
+        public SocialMediaLink(string title, string slug, string url, string icon, string platform)
```
Comment (lydiagray) [inline src/StockportWebapp/Models/SocialMediaLink.cs:11]: This should be Account Name not Platform

File: src/StockportWebapp/Models/SocialMediaLink.cs [hunk 8]
```diff
@@ -6,21 +6,23 @@ public class SocialMediaLink
...
+            Platform = platform;
         }
     }
 
     public class NullSocialMediaLink : SocialMediaLink
     {
         public NullSocialMediaLink()
             : base(
-                string.Empty, string.Empty, string.Empty, string.Empty)
+                string.Empty, string.Empty, string.Empty, string.Empty, string.Empty)
```
Comment (lydiagray) [inline src/StockportWebapp/Models/SocialMediaLink.cs:25]: As above

**Pull request merged**